File: app/src/main/java/net/squanchy/search/view/HeaderType.java
package net.squanchy.search.view;

import android.support.annotation.StringRes;

import net.squanchy.R;

public enum HeaderType {
    SPEAKERS(R.string.speaker_list_title),
    EVENTS(R.string.talks_list_title),
    TRACKS(R.string.speaker_list_title);

    @StringRes
    private final int headerTextResourceId;

    HeaderType(@StringRes int headerTextResourceId) {
        this.headerTextResourceId = headerTextResourceId;
    }

    @StringRes
    public int headerTextResourceId() {
        return headerTextResourceId;
    }
}

File: app/src/main/java/net/squanchy/search/view/HeaderType.kt
package net.squanchy.search.view

import androidx.annotation.StringRes

import net.squanchy.R

enum class HeaderType(
    @param:StringRes
    @get:StringRes
    val headerTextResourceId: Int
) {
    SPEAKERS(R.string.speaker_list_title),
    EVENTS(R.string.talks_list_title),
    TRACKS(R.string.speaker_list_title)
}
